python/langsmith/client.py

```diff
@@ -742,6 +746,14 @@ def _content_above_size(self, content_length: Optional[int]) -> Optional[str]:
         if size_limit is None:
             return None
         if content_length > size_limit:
+            should_debug_crash_dump = os.getenv("LANGSMITH_DEBUG_CRASH_DUMP") in [
+                "1",
+                "true",
+            ]
+            if should_debug_crash_dump and request is not None:
+                with open("content_size_limit_crash_dump.jsonl", "a") as f:
```

Comment:
need to specify `'utf-8'` as the encoding.
Also, let's use gzip - by definition these will get pretty big pretty fast